packages/@react-aria/datepicker/src/useDatePicker.ts

```diff
@@ -71,8 +72,39 @@ export function useDatePicker<T extends DateValue>(props: AriaDatePickerProps<T>
   let domProps = filterDOMProps(props);
   let focusManager = useMemo(() => createFocusManager(ref), [ref]);
 
+  let {focusWithinProps} = useFocusWithin({
+    ...props,
+    onBlurWithin: (e: FocusEvent) => {
+      if (state.isOpen) {
+        return;
+      }
+
+      if (props.onBlur) {
+        props.onBlur(e);
+      }
+    },
+    onFocusWithin: (e: FocusEvent) => {
+      if (state.isOpen) {
+        return;
+      }
+
+      if (props.onFocus) {
+        props.onFocus(e);
+      }
+    },
+    onFocusWithinChange: (value: boolean) => {
+      if (state.isOpen) {
+        return;
+      }
+
+      if (props.onFocusChange) {
+        props.onFocusChange(value);
+      }
+    }
```

Comment:
An easier way to do this could be

```js
useFocusWithin({
   ...props,
  isDisabled: state.isOpen,
  onBlurWithin: props.onBlur,
  // ...
});
```